### PR TITLE
run integration tests as part of the GHA matrix

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: .github/workflows/foreman.yml [hunk 1]
```diff
@@ -33,7 +34,15 @@ jobs:
...
-      matrix: ${{ fromJson(needs.setup_matrix.outputs.matrix) }}
+      matrix:
+        ruby: ${{ fromJson(needs.setup_matrix.outputs.matrix).ruby }}
+        node: ${{ fromJson(needs.setup_matrix.outputs.matrix).node }}
+        postgresql: ${{ fromJson(needs.setup_matrix.outputs.matrix).postgresql }}
+        task:
+          - 'test:units'
+          - 'test:functionals'
+          - 'test:graphql'
+          - 'jenkins:integration'
```
Comment (ekohl) [inline .github/workflows/foreman.yml]: Hacking the planet: can avoid the Jenkins task and simply use a space for multiple tasks?
```suggestion
          - 'webpack:compile test:integration'
```

Alternatively we can create a `webpack:compile_if_needed` task and enhance `test:integration` (which I think comes from Rails itself) with it as a dependency.

Comment (evgeni) [reply .github/workflows/foreman.yml]: yes, that'd work

File: .github/workflows/foreman.yml [hunk 2]
```diff
@@ -53,13 +62,31 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node }}
+      - name: Generate package-lock.json
+        run: npm install --package-lock-only --no-audit
+      - name: Setup Cache
```
Comment (evgeni) [inline .github/workflows/foreman.yml]: this gives exactly 0 speedup

Comment (ekohl) [reply .github/workflows/foreman.yml]: I also had the impression the cache was really not intended to be used without a package lock file in your repository.

Comment (evgeni) [reply .github/workflows/foreman.yml]: Yeah. But I did create a lock, and it did show a cache hit in the logs, and yet the install step is 65 seconds long in cache and no cache cases :woman_shrugging:

Comment (evgeni) [reply .github/workflows/foreman.yml]: mhh, with `npm ci` it's now 45s, which is better, I guess?

Comment (evgeni) [reply .github/workflows/foreman.yml]: the benchmark, without any caching, was 95s, now it's 45s+35s (for the lock generation) = 80s. Not that much of a win :(

Comment (ekohl) [reply .github/workflows/foreman.yml]: Technically we only need NPM packages installed for integration testing. It's tempting to skip it for other tasks.

And I wonder if we could cache the `package-lock.json` file based on the `package.json` file hash.

Comment (evgeni) [reply .github/workflows/foreman.yml]: interesting observation: lock + install with caches takes 70 secs (after restoring a "close enough" cache based on package.json alone), while "install w/o locking first" takes 80secs with the same cache. No idea why.

Comment (evgeni) [reply .github/workflows/foreman.yml]: one thing that we could speed up (but I am not sure I like) is to cache `node_modules` itself, as the extraction takes quite some time

**Comment by evgeni** [comment]
I think this is good enough now

File: .github/workflows/foreman.yml [hunk 3]
```diff
@@ -87,6 +88,26 @@ jobs:
...
         with:
           node-version: ${{ matrix.node }}
+      - name: Setup NPM Cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ matrix.node }}-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.node }}-
+        if: contains(matrix.task, 'compile')
```
Comment (ekohl) [inline .github/workflows/foreman.yml:98]: Oh nice one.

**Review by ekohl — APPROVED** [review]

**Pull request merged**